packages/react/focus-scope/src/FocusScope.tsx

```diff
@@ -103,18 +103,21 @@ const FocusScope = React.forwardRef((props, forwardedRef) => {
   React.useEffect(() => {
     if (container) {
       focusScopesStack.add(focusScope);
-      const PREVIOUSLY_FOCUSED_ELEMENT = document.activeElement as HTMLElement | null;
-
-      // we need to setup the listeners before we `dispatchEvent`
-      container.addEventListener(AUTOFOCUS_ON_MOUNT, onMountAutoFocus);
-      container.addEventListener(AUTOFOCUS_ON_UNMOUNT, onUnmountAutoFocus);
-
-      const mountEvent = new Event(AUTOFOCUS_ON_MOUNT, EVENT_OPTIONS);
-      container.dispatchEvent(mountEvent);
-      if (!mountEvent.defaultPrevented) {
-        focusFirst(getTabbableCandidates(container), { select: true });
-        if (document.activeElement === PREVIOUSLY_FOCUSED_ELEMENT) {
-          focus(container);
+      const previouslyFocusedElement = document.activeElement as HTMLElement | null;
+      const hasAutoFocusedCandidate = container.contains(previouslyFocusedElement);
```

Comment:
I get how this works because of the order it's run, but perhaps we should be more explicit by checking that it's inside and that it has `autofocus` attribute?

Reply:
Interesting, perhaps I should rename this to `hasFocusedCandidate` instead because it doesn't necessarily need to be tied to `autoFocus` attribute. 

If something has already been focused inside by any means at the time that it opens (e.g. programatically) then we don't need to auto focus anything ourselves.

Reply:
Yeah I guess that would make sense.